### PR TITLE
ci: run windows CI jobs on PRs

### PR DESCRIPTION
**Pull request by josephperrott**
Previously windows CI jobs were only run on upstream branches, with the addition
of larger Windows executors as well as the improvement of setup speed in the
windows environment setup script allows for the windows tests to pass in a
reasonable timeframe.

### CODE REVIEW

File: .circleci/windows-env.ps1 [hunk 1]
```diff
@@ -2,8 +2,8 @@
 # https://docs.bazel.build/versions/master/install-windows.html
 # https://docs.bazel.build/versions/master/windows.html
 # Install MSYS2 and packages
-choco install msys2 --version 20180531.0.0 --no-progress --package-parameters "/NoUpdate"
-C:\tools\msys64\usr\bin\bash.exe -l -c "pacman --needed --noconfirm -S zip unzip patch diffutils git"
+choco install msys2 --version 20200903.0.0 --no-progress --package-parameters "/NoUpdate"
```
Comment (sonukapoor) [inline .circleci/windows-env.ps1:5]: Does `choco` automatically install `msys2` into the `C:\tools\msys64 folder`? I have it installed in my `C` folder only.

Comment (josephperrott) [reply .circleci/windows-env.ps1:5]: It installs to the value of `Get-ToolsLocation`, which at least on CircleCI is `C:\tools`

Comment (sonukapoor) [reply .circleci/windows-env.ps1:5]: Got it 👍

File: .circleci/config.yml [hunk 2]
```diff
@@ -184,22 +182,12 @@ commands:
     description: Setup windows node environment
     steps:
       # Use the Linux workspace directly, as it already has checkout, rebased and node modules.
```
Comment (gkalpak) [inline .circleci/config.yml:184]: This comment is no longer valid.

**Review by gkalpak — APPROVED** [review]
🚀

**Review by IgorMinar — APPROVED** [review]
Overall this looks good except that in the last CI run the windows VE job took 20min (most of all the jobs) so it seems that this job will become a bottleneck for the CI.

Interestingly, the 20min long job run already benefited from many cache hits, which means that on average the job duration might be even worse.

Let's get this PR in, but let's also monitor the job duration to understand the overall impact of this change and consider further changes to the windows setup if necessary.

**Pull request closed without merging**